engine/player/unique_gear_shadowlands.cpp

```diff
@@ -4280,80 +4273,172 @@ void yasahm_the_riftbreaker( special_effect_t& effect )
+      : proc_spell_t( "sadistic_glee", e.player, e.player->find_spell( 353466 ), e.item ),
+        scaled_dmg( e.driver()->effectN( 1 ).average( e.item ) )
     {
       base_td = scaled_dmg;
     }
 
-    double base_ta(const action_state_t* /* s */) const override
+    double base_ta( const action_state_t* /* s */ ) const override
     {
       return scaled_dmg;
     }
   };
 
-  auto sadistic_glee = static_cast<sadistic_glee_t*>(effect.player->find_action("sadistic_glee"));
+  auto sadistic_glee = static_cast<sadistic_glee_t*>( effect.player->find_action( "sadistic_glee" ) );
 
-  if (!sadistic_glee)
-    effect.execute_action = create_proc_action<sadistic_glee_t>("sadistic_glee", effect);
+  if ( !sadistic_glee )
+    effect.execute_action = create_proc_action<sadistic_glee_t>( "sadistic_glee", effect );
   else
-    sadistic_glee->scaled_dmg += effect.driver()->effectN(1).average(effect.item);
+    sadistic_glee->scaled_dmg += effect.driver()->effectN( 1 ).average( effect.item );
 
   effect.spell_id = 357588;
   /* apparently due to proc rate being lower than reported in spell data (?) - emallson */
   effect.rppm_modifier_ = 0.5;
 
-
   /* override proc rate for tanks (40% of regular proc rate unless option is
      set), and allow override via expansion option */
   auto proc_option = effect.player->sim->shadowlands_opts.cruciform_veinripper_proc_rate;
-  if (proc_option == 0.0 && effect.player->position() == POSITION_FRONT) {
+  if ( proc_option == 0.0 && effect.player->position() == POSITION_FRONT )
+  {
     effect.rppm_modifier_ *= 0.4;
-  } else if(proc_option > 0.0) {
+  }
+  else if ( proc_option > 0.0 )
+  {
     effect.rppm_modifier_ *= proc_option;
   }
 
-  new dbc_proc_callback_t(effect.player, effect);
+  new dbc_proc_callback_t( effect.player, effect );
 }
 
-void singularity_supreme( special_effect_t& effect )
+struct singularity_supreme_t : public stat_buff_t
 {
-  auto lockout = make_buff( effect.player, "singularity_supreme_lockout", effect.player->find_spell( 368865 ) )
-    ->set_quiet( true );
+  stat_buff_t* supremacy_buff;
+  buff_t* lockout;
+  stat_buff_t* swap_stat_compensation;
+
+  singularity_supreme_t( player_t* p, const item_t* i )
+    : stat_buff_t( p, "singularity_supreme_counter", p->find_spell( 368845 ), i )
+  {
+    lockout = buff_t::find( p, "singularity_supreme_lockout" );
+    if ( !lockout )
+    {
+      lockout = make_buff( p, "singularity_supreme_lockout", p->find_spell( 368865 ) )
+                    ->set_quiet( true )
+                    ->set_can_cancel( false );
+    }
+
+    supremacy_buff = dynamic_cast<stat_buff_t*>( buff_t::find( p, "singularity_supreme" ) );
+    if ( !supremacy_buff )
+    {
+      supremacy_buff = make_buff<stat_buff_t>( p, "singularity_supreme", p->find_spell( 368863 ), i );
+      supremacy_buff->set_can_cancel( false );
+    }
+
+    swap_stat_compensation = dynamic_cast<stat_buff_t*>( buff_t::find( p, "swap_stat_compensation" ) );
```

Comment:
I don't think there is any need to find or create this buff when the option to enable swapping is not set.